server/refishing-fst2.txt

```diff
@@ -8,13 +8,25 @@ clear
 push Dutch
 save stack dutch.bin
 
-# English
-define English [] ;
+# English (Needs fixing)
+def SK "s k " -> "ʃ ";
+def XS "x s " -> "k s ";
+def XH "x " -> "h ";
+def KN "k n " -> "n ";
+
+# hacks
+def AR "a r " -> "ɜ r ";
+def ER "ɜ r " -> "ɜː ";
+
+define English PGWord .o. SK .o. AR .o. ER;
```

Comment:
@LinguList Nathan and I are a bit confused why this doesn't work, and maybe you have some input. The rules work fine, but the PGWord seems to cause certain words not to pass through the transducer when applying upwards. However, if we remove PGWord then we run into issues with foma reading certain characters.

Reply:
I also did not understand it, and it did neither work in German, when I tried something similar.

I understood it works if you write:

```
regex PGWord;
regex SK .o. AR .o. ER;
```
But dunno why it would not work with `.o.` for all statements.

Reply:
Maybe, the spaces after all confuse foma? But I had similar issues with other symbols, and there seems to be a grammatical error from our side when it comes to the concatenation, that we do not see here some how. At least that's what I concluded, leaving it for now, as I felt I cannot solve it easily without reading more about foma syntax (but the tutorial is sparse).

Reply:
Interesting, looks like this will be something to look into more then.

Reply:
I can confirm that up and down basically works with the following order, which I wrote into a text file:

```
# German
def PGC "b " | "d " | "f " | "g " | "gʷ " | "j " | "k " | "kʷ " | "kː " | "l " | "lː " | "m " | "mː " | "n " | "nː " | "p " | "pː " | "r " | "s " | "t " | "w " | "x " | "xʷ " | "z " | "θ ";

def PGV "a " | "ai " | "au " | "e " | "eu " | "eː " | "i " | "iː " | "o
" | "oː " | "u " | "uː ";

def PGSyllable PGC* PGV* ;

def PGWord PGSyllable* ;

def SK "s k " -> "ʃ ";
def XS "x s " -> "k s ";
def XH "x " -> "h ";
def KN "k n " -> "n ";

regex PGWord;
regex SK .o. XS .o. XH .o. KN ;
```

Reply:
Foma interacticelyis then triggered:

```
foma -l data.txt
```

Reply:
Then, typing `up`, you can insert words, I did this also to test the backwards reconstruction.

```
apply up> ʃ a k s 
s k a k s 
s k a x s 
ʃ a k s 
ʃ a x s 
```

Reply:
I think there's a problem in concatenating rudimentary word definition for proto-germaic, which -- similar to the burmish word definition by Xun -- defines the ancestral language, and the insertion of new replacement rules into this transducer.

Reply:
If I combine it in one regex as `PGWord .o. XS` etc., it does not work anymore.

Reply:
Update, @knightss27, it works if you separate the spaced instances, so you write each sound as `"s "`.

Check this:

```
# German
def PGC "b " | "d " | "f " | "g " | "gʷ " | "j " | "k " | "kʷ " | "kː " | "l " | "lː " | "m " | "mː " | "n " | "nː " | "p " | "pː " | "r " | "s " | "t " | "w " | "x " | "xʷ " | "z " | "θ " ;

def PGV "a " | "ai " | "au " | "e " | "eu " | "eː " | "i " | "iː " | "o
" | "oː " | "u " | "uː " ;

def PGSyllable PGC* PGV* ;

def PGWord PGSyllable* ;

def SK "s " "k " -> "ʃ " ;
def XS "x " "s " -> "k s " ;
def XH "x " -> "h " ;
def KN "k n " -> "n " ;

def German PGWord .o. SK .o. XS .o. XH .o. KN ;
regex German ;
```
It works perfectly.

Reply:
So the confusing part here is that the space seems to cause an error inside the `"` quotes, unless you really treat each char as a single unit. Which is easy to do.

Reply:
AND I even know the reason why: since foma has a certain way of defining segments, you must be careful in defining longer units that end up in the network as one char, which conflict with smaller units.

To replicate, try this:

```
def A aa;
def AB a -> b;
```
This conflicts, since you define `aa` as one symbol, you cannot invoke a as a single symbol anymore !

Reply:
So one must treat all instances of segment + space as one unit. And one must write them next to each other:

```
"s " "k " "a "
```

not 

```
"s k a "
```